python/tank_vendor/shotgun_deploy/io_descriptor/git_branch.py

```diff
@@ -53,14 +53,16 @@ def __init__(self, descriptor_dict):
         :param descriptor_dict: descriptor dictionary describing the bundle
         :return: Descriptor instance
         """
-        super(IODescriptorGitBranch, self).__init__(descriptor_dict)
-
+        # make sure all required fields are there
         self._validate_descriptor(
             descriptor_dict,
             required=["type", "path", "version", "branch"],
             optional=[]
         )
 
+        # call base class
+        super(IODescriptorGitBranch, self).__init__(descriptor_dict)
```

Comment:
The `IODescriptorGitBranch` tries to access "path" so we should validate before this constructor is run.